core/store/src/trie/shard_tries.rs

```diff
@@ -590,66 +655,40 @@ impl ShardTries {
                 state_snapshot_subdir,
                 compaction_enabled: _,
             } => {
-                let path = Self::get_state_snapshot_base_dir(
-                    &CryptoHash::new(),
+                // directly return error if no snapshot is found
+                let snapshot_hash: CryptoHash = self.get_state_snapshot_hash()?;
+
+                let snapshot_path = Self::get_state_snapshot_base_dir(
+                    &snapshot_hash,
                     &home_dir,
                     &hot_store_path,
                     &state_snapshot_subdir,
                 );
-                let parent_path =
-                    path.parent().ok_or(anyhow::anyhow!("{path:?} needs to have a parent dir"))?;
-                tracing::debug!(target: "state_snapshot", ?path, ?parent_path);
-
-                let snapshots = match std::fs::read_dir(parent_path) {
-                    Err(err) => {
-                        if err.kind() == std::io::ErrorKind::NotFound {
-                            tracing::debug!(target: "state_snapshot", ?parent_path, "State Snapshot base directory doesn't exist.");
-                            return Ok(());
-                        } else {
-                            return Err(err.into());
-                        }
-                    }
-                    Ok(entries) => {
-                        let mut snapshots = vec![];
-                        for entry in entries.filter_map(Result::ok) {
-                            let file_name = entry.file_name().into_string().map_err(|err| {
-                                anyhow::anyhow!("Can't display file_name: {err:?}")
-                            })?;
-                            if let Ok(prev_block_hash) = CryptoHash::from_str(&file_name) {
-                                snapshots.push((prev_block_hash, entry.path()));
-                            }
-                        }
-                        snapshots
-                    }
-                };
+                let parent_path = snapshot_path
```

Comment:
No longer needed?

Reply:
I think we can keep this, so we know that this function erred out because parent directory does not even exist.